dogsdogsdogs/src/operators/lookup_map.rs

```diff
@@ -101,7 +101,7 @@ where
                                 while let Some(value) = cursor.get_val(&storage) {
                                     let mut count = Tr::Diff::zero();
                                     cursor.map_times(&storage, |t, d| {
-                                        if t.less_equal(time) { count.plus_equals(d); }
+                                        if t.into_owned().less_equal(time) { count.plus_equals(&d.into_owned()); }
```

Comment:
Mentioned elsewhere but in further detail: 
1. I think we have the ability to set up `PartialOrder` for success, because the LHS is a type we control. The RHS may be .. `isize` or something like it, and we are likely unable to get a `PartialOrder` implementation for it (due to coherence). But there are some missing implementations for `PartialOrder` before this works easily.
2. I'm less sure we can get `Semigroup` working, as when it is `isize` it probably doesn't know to implement e.g. `Semigroup<DiffGat<'_>>`. Perhaps that means that for naive types your GAT must be the type itself (seems fine, tbh) and for complex types you need to make sure to provide such an implementation. But, more dangerous because the LHS is less obviously under our control under idiomatic use.

Reply:
This `into_owned().less_equal(time)` is one of the few `into_owned()` moments that remain that feel bad. It was not previously allocating, and is now allocating. I can imagine we may find more, but one can always recover the original behavior by requiring the GAT to be `&Something`.